### PR TITLE
Adds phpunit to dev-tools

### PR DESCRIPTION
**Pull request by vincent-gao**
### Motivation
1. we would like standardised unit tests for our modules, we raised this requirement in a dev meeting in last year.
2. it is part of my PDP 😬 

actual unit tests should reside in each tide_* modules.

ok, it runs under CI. 
it is ready to roll out to all tide_* modules. and this update focuses `unit` test only.

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+         beStrictAboutOutputDuringTests="true"
+         beStrictAboutChangesToGlobalState="true"
+         printerClass="\Drupal\Tests\Listeners\HtmlOutputPrinter">
+  <php>
+    <!-- Set error reporting to E_ALL. -->
+    <ini name="error_reporting" value="32767"/>
+    <!-- Do not limit the amount of memory tests take to run. -->
+    <ini name="memory_limit" value="-1"/>
+    <!-- Example SIMPLETEST_BASE_URL value: http://localhost -->
+    <env name="SIMPLETEST_BASE_URL" value=""/>
```
Comment (vincent-gao) [inline phpunit.xml.dist:19]: I have tried both `http://localhost` and `http://nginx`, neither works. 
my question is that , what is correct value for this ,  is it possible to access `base url` from the container.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    <!-- Example SIMPLETEST_DB value: mysql://username:password@localhost/databasename#table_prefix -->
+    <env name="SIMPLETEST_DB" value=""/>
```
Comment (vincent-gao) [inline phpunit.xml.dist:21]: I guess this one should be `mysql://drupal:drupal@mariadb/drupal`?

File: phpunit.xml.dist (hunk 1 above)
Comment (GROwen) [reply phpunit.xml.dist:19]: The value should be whatever https://github.com/dpc-sdp/dev-tools/blob/b6b250005595167288e7908f2792d6090eabfd1c/.env#L35 is set to.

File: phpunit.xml.dist (hunk 2 above)
Comment (GROwen) [reply phpunit.xml.dist:21]: I _think_ it's the same as https://github.com/dpc-sdp/dev-tools/blob/b6b250005595167288e7908f2792d6090eabfd1c/.env#L35

i.e. `drupal:drupal@mysite.docker.amazee.io/drupal`

File: phpunit.xml.dist (hunk 1 above)
Comment (vincent-gao) [reply phpunit.xml.dist:19]: Thanks @GROwen !  yeah , we can access it from the outside of the container, but what if I would like to access it from containers like `php` or `cli` container?

Comment (vincent-gao) [reply phpunit.xml.dist]: I think , I just missed the port number.

**Review by anthony-malkoun — CHANGES_REQUESTED** [review]
Hey Vincent can you add a little bit of explanation to the README.md, e.g. how to run the tests manually etc?

File: README.md [hunk 3]
```diff
@@ -118,6 +118,7 @@ Finally, we install the module itself.
 1. Uncomment this line in `docker-compose.yml`:
   ```
   #XDEBUG_ENABLE: "true"
+  #DOCKERHOST: "host.docker.internal"
```
Comment (vincent-gao) [inline README.md:121]: Just found it missed `DOCKERHOST` variable

**Comment by vincent-gao** [comment]
> Hey Vincent can you add a little bit of explanation to the README.md, e.g. how to run the tests manually etc?

Hey @anthony-malkoun , I have updated README.md file . please take a look. Thanks

File: README.md [hunk 4]
```diff
@@ -181,3 +182,14 @@ To make a change:
 6. If the build passes, merge the Dev Tools branch to `master` and make a release.
     - NOTE: This must be a release in GitHub, see https://github.com/dpc-sdp/dev-tools/releases/new
 7. Remove the temporary branch in `tide_*` module from Step 3. 
+
```
Comment (anthony-malkoun) [inline README.md:185]: 😍 Thanks for doing this Vincent.

**Review by anthony-malkoun — APPROVED** [review]

**Pull request merged**